sockeye/training.py

```diff
@@ -611,6 +634,13 @@ def fit(self,
                     logger.info("Validation-%s has not improved for %d checkpoints, best so far: %f",
                                 early_stopping_metric, self.state.num_not_improved, self.state.best_metric)
 
+                # (5) detect divergence with respect to perplexity value at the last checkpoint
+                if self.state.metrics and not has_improved:
+                    # perplexity cannot exceed the number of target words, using the double to avoid precision issues
+                    last_ppl_value = self.state.metrics[-1]["%s-val" % C.PERPLEXITY]
+                    if not np.isfinite(last_ppl_value) or last_ppl_value > 2 * len(self.target_vocab):
+                        self.state.diverged = True
```

Comment:
I think when we detect that the model diverged a logging message would be appropriate.

Reply:
Added